tests/integration/nm/vxlan_test.py

```diff
@@ -0,0 +1,127 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import pytest
+
+from distutils.version import StrictVersion
+
+from libnmstate import nm
+from libnmstate.schema import Interface
+from libnmstate.schema import VXLAN
+
+from .testlib import mainloop_run
+
+
+def test_create_and_remove_vxlan(eth1_up):
+    vxlan_desired_state = _create_vxlan_state(eth1_up)
+    with _vxlan_interface(vxlan_desired_state):
+        vxlan_name = _vxlan_ifname(vxlan_desired_state)
+        vxlan_current_state = _get_vxlan_current_state(vxlan_name)
+        assert vxlan_desired_state == vxlan_current_state
+
+    assert not _get_vxlan_current_state(vxlan_name)
+
+
+@pytest.mark.xfail(
+    condition=StrictVersion(nm.nmclient.nm_version()) < StrictVersion('1.22'),
+    strict=True,
+    reason='https://bugzilla.redhat.com/show_bug.cgi?id=1768388',
+)
+def test_read_destionation_port_from_libnm(eth1_up):
+    vxlan_desired_state = _create_vxlan_state(eth1_up)
+    with _vxlan_interface(vxlan_desired_state):
+        vxlan_name = vxlan_desired_state[Interface.KEY][0][Interface.NAME]
```

Comment:
This seems to be wrong.
The vxlan_desired_state is pointing to the vxlan subtree, not to a full state.